os/fs/smartfs/smartfs_smart.c

```diff
@@ -255,6 +256,14 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 			}
 		}
 
+		for (ofile_ptr = fs->fs_head; ofile_ptr != NULL; ofile_ptr = ofile_ptr->fnext) {
```

Comment:
it just implemented what I told you but it should allow if flag is read-only.
so when it commit changes, it should parse sf again and change offset & sector of sf which reading same entry.
Of course you can keep current status for trying but my comment need to be considered. there are many use-case write & read happened in different thread at the same time especially streaming case.

Reply:
All use cases where different threads/different fds open the same file are included here. 2 instances cannot be opened at the same time.
What is your idea of blocking second fd.
According to the durability principle, if one operation returns success, it's changes should be visible in the next query. But if we hold some data in the buffer, then the changes are not committed to flash. Hence I think blocking is fine till the file  is closed.
It is the responsibility of the app developer to make sure that the file is closed for syncing as soon as the operations are complete.